web/src/components/welcome-modal/welcome-modal.tsx

```diff
@@ -22,8 +25,28 @@ export interface WelcomeModalProps extends ModalProps {
 export default ({ challengeToken, teamToken, ...props }: WelcomeModalProps) => {
   const readableTeamName = challengeTeams[teamToken].readableName;
   const [hasAgreed, setHasAgreed] = useState<boolean>(false);
+  const account = useAccount();
+  const saveAccount = useAction(User.actions.saveAccount);
+
   useEffect(() => trackChallenge('modal-welcome'), []);
 
+  const parseEnrollment = (queryString: string): Enrollment => {
+    const regex = new RegExp(/([a-z]+)=([a-z]+)/, 'gm');
+    const queries = {} as { [key: string]: string };
+    let pair: Array<string> = [];
+
+    while ((pair = regex.exec(queryString)) !== null) {
+      queries[pair[1]] = pair[2];
+    }
+
+    return {
+      challenge: queries.challenge,
+      team: queries.team,
+      referer: queries.referer,
+      invite: queries.invite,
+    } as Enrollment;
+  };
```

Comment:
This whole method is messy and I don't love it, very much open to suggestions for improvements. Without using an external library for accessing query params, this was my best shot. Also, the casting at `return` seems pretty redundant but that was the best way I could think of to avoid a meaningless `:Object` return type.

Reply:
You could always use this piece of garbage we coded together last week 😂 https://gist.github.com/rileyjshaw/32d7455e63c129d6064d406870a427c2

...but I like this, it's clear. Check the comments above though, I think it's not working for invite codes currently.

Also re: the casting at `return`, if we wanted to do some safety checks we could do something like:

```
if (challengeTokens.includes(queries.challenge) && challengeTeamTokens.includes(queries.team)) {
    return queries as Enrollment;
} else return null;
```

Reply:
The thing I was trying to guard against is if `challengeTokens` has invalid properties beyond the four that `Enrollment` defines for some reason - it's probably overly cautious and I think (I could be wrong?) it would only generate a warning, not an error, so this is probably fine for prod.

Reply:
Ah, makes sense. In that case, it looks like we have access to a `pick` function:

```javascript
const pick = require('lodash.pick');
// ...
if (challengeTokens.includes(queries.challenge) && challengeTeamTokens.includes(queries.team)) {
    return pick(queries, 'challenge', 'team', 'invite') as Enrollment;
} else return null;
```